includes/class-dispatcher.php

```diff
@@ -87,6 +88,48 @@ public function _execute_view( $template ) {
 		return $template;
 	}
 
+    /**
+     * Redirect from draft listings to the main directory page.
+     *
+     * @since x.x.x
+     *
+     * @return void
+     */
+    public function maybe_redirect_from_draft_listings() {
+        global $wp_query, $post, $wpdb;
+
+        if ( $post !== NULL ) {
+            return;
+        }
+
+        $vars = $wp_query->query_vars;
+
+        if ( ! isset( $vars['post_type'] ) || $vars['post_type'] !== WPBDP_POST_TYPE ) {
+            return;
+        }
+
+        // We run a custom query here because we want to check for draft/pending listings and WP_Query doesn't return them.
+        $post = $wpdb->get_results(
+            $wpdb->prepare(
+                "
+                SELECT * FROM $wpdb->posts
```

Comment:
Is there a reason not to use get_posts()?

Reply:
I was definitely doing something wrong and not realizing what because when I worked on it first no matter what I tried with `get_posts` it was always returning an empty array so eventually I went for a custom query.

But now when I tried it again after a few hours break it worked so it was just me being too tired and making silly mistakes. 

I changed that to using `get_posts`. Thanks @stephywells!